src/main/java/mate/academy/dao/impl/MovieDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package mate.academy.dao.impl;
+
+import mate.academy.dao.MovieDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    public static final SessionFactory SESSION_FACTORY = HibernateUtil.getSessionFactory();
+
+    @Override
+    public Movie add(Movie movie) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = SESSION_FACTORY.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add movie to DB " + movie, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Movie get(Long id) {
+        Session session = SESSION_FACTORY.openSession();
+        return session.get(Movie.class, id);
```

Comment:
Seems like we should use TWR block here and return Optional

  public interface MovieDao {
      Movie add(Movie movie);
  
      Optional<Movie> get(Long id);
  }